rust/datafusion/src/physical_plan/planner.rs

```diff
@@ -42,14 +44,30 @@ use crate::physical_plan::{AggregateExpr, ExecutionPlan, PhysicalExpr, PhysicalP
 use arrow::compute::SortOptions;
 use arrow::datatypes::Schema;
 
+/// This trait permits the `DefaultPhysicalPlanner` to create plans for
+/// user defined `ExtensionPlanNode`s
+pub trait ExtensionPlanner {
```

Comment:
@andygrove  -- here is my  proposed change to support physical planning of extension nodes in response to https://github.com/apache/arrow/pull/8020#discussion_r475953168. Rather than doing physical planning directly in the `ExtensionNode` itself, this is a mechanism to register planning capabilities into the `DefaultPhysicalPlanner`